src/jsc/jsc_function.hpp

```diff
@@ -29,6 +29,17 @@ inline JSValueRef jsc::Function::call(JSContextRef ctx, const JSObjectRef& funct
 {
     JSValueRef exception = nullptr;
     JSValueRef result = JSObjectCallAsFunction(ctx, function, this_object, argc, arguments, &exception);
+
+    // Flush the React Native UI task queue whenever we call into JS from C++ – see `_flushUiTaskQueue` in
```

Comment:
I wonder if it makes sense to move this to an `inline`d function that you call from [`jsc::Function::construct`](https://github.com/realm/realm-js/pull/4330/files#R57) too? It might potentially also schedule async work.